docs/memcached.md

```diff
@@ -0,0 +1,54 @@
+# `memcached` Metrics Receiver
+
+The memcached receiver can retrieve stats from your memcached server through the . 
+
+
+## Configuration
+
+Following the guide for [Configuring the Ops Agent](https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/configuration#file-location), add the required elements for your memcached configuration.
+
+To configure a receiver for your memcached metrics, specify the following fields:
+
+| Field                 | Default                   | Description |
+| ---                   | ---                       | ---         |
+| `type`                | required                  | Must be `memcached`. |
+| `endpoint`            | `localhost:3306`          | The url, or unix socket file path, for your memcached server. |
+| `collection_interval` | `60s`                     | A [time.Duration](https://pkg.go.dev/time#ParseDuration) value, such as `30s` or `5m`. |
+
+Example Configuration:
+
+```yaml
+metrics:
+  receivers:
+    redis_metrics:
+      type: memcached
+      address: localhost:3306
+      collection_interval: 30s
+  service:
+    pipelines:
+      memcached_pipeline:
+        receivers:
+          - memcached_metrics
+```
+
+## Metrics
+
+The Ops Agent collects the following metrics from your memcached servers.
+
+| Name | Description | Unit | Type | Attributes |
+| ---- | ----------- | ---- | ---- | ---------- |
+| memcached.bytes | Current number of bytes used by this server to store items. | By | Gauge | <ul> </ul> |
+| memcached.commands | Commands executed. | 1 | Sum | <ul> <li>command</li> </ul> |
+| memcached.current_connections | The current number of open connections. | connections | Gauge | <ul> </ul> |
+| memcached.current_items | Number of items currently stored in the cache. | 1 | Gauge | <ul> </ul> |
+| memcached.evictions | Cache item evictions. | 1 | Sum | <ul> </ul> |
+| memcached.network | Bytes transferred over the network. | by | Sum | <ul> <li>direction</li> </ul> |
+| memcached.operation_hit_ratio | Hit ratio for operations, expressed as a percentage value between 0.0 and 100.0. | % | Gauge | <ul> <li>operation</li> </ul> |
+| memcached.operations | Operation counts. | 1 | Sum | <ul> <li>type</li> <li>operation</li> </ul> |
+| memcached.rusage | Accumulated user and system time. | 1 | Sum | <ul> <li>state</li> </ul> |
```

Comment:
What are the values for state? `user` and `system`?

Is `.rusage` the OT metric convention? (I assume it's short for "resource usage"?)

Reply:
it's how memcached [references it](https://docs.oracle.com/cd/E17952_01/mysql-5.6-en/ha-memcached-stats.html). and yes `user` and `system`, I'll add a table for attribute values.

Reply:
Ah, in the link memcached calls it `rusage_user` and `rusage_system` - in that context the name makes sense. But if you're just calling it `rusage` I thiink that is too generic - [`getrusage`](https://man7.org/linux/man-pages/man2/getrusage.2.html) is a system call that returns CPU usage, memory usage, and a number of other stats. Maybe we could call it `memcached.rusage.cpu`?

Reply:
ok will update.

Reply:
Updated to `memcached.cpu.usage` based on what otel wanted. is that fine for google or do we need to change it for ops-agent?

Reply:
That name is fine with me